trs_cli/client.py

```diff
@@ -88,6 +94,81 @@ def __init__(
     #
     # Check DRS-cli repo for examples
 
+    def get_descriptor(
+        self,
+        type: str,
+        id: str,
+        version_id: str,
+        token: Optional[str] = None
+    ) -> Union[FileWrapper, Error]:
+        """Get the tool descriptor for the specified tool.
+
+        Arguments:
+            type: The output type of the descriptor. Plain types return
+                the bare descriptor while the "non-plain" types return a
+                descriptor wrapped with metadata. Allowable values include
+                "CWL", "WDL", "NFL", "GALAXY", "PLAIN_CWL", "PLAIN_WDL",
+                "PLAIN_NFL", "PLAIN_GALAXY".
+            id: A unique identifier of the tool, scoped to this registry.
+            version_id: An identifier of the tool version, scoped to this
+            registry.
+            token: Bearer token for authentication. Set if required by TRS
+                implementation and if not provided when instatiating client or
+                if expired.
+
+        Returns:
+            Unmarshalled TRS response as either an instance of
+            `FileWrapper` in case of a `200` response, or an instance of
+            `Error` for all other JSON reponses.
+
+        Raises:
+            requests.exceptions.ConnectionError: A connection to the provided
+                DRS instance could not be established.
+            trs_cli.errors.InvalidResponseError: The response could not be
+                validated against the API schema.
+        """
+        url = f"{self.uri}/tools/{id}/versions/{version_id}/{type}/descriptor"
+        if token:
+            self.token = token
+            self._get_headers()
```

Comment:
Can you call `_get_headers()` _all the time_, not just if a token was passed? This needs to change in DRS-cli as well, but it's not urgent. Also, there is a mistake in `_get_headers()`, both in here and DRS-cli: the header that is always to be set, token or not, is `Accept`, not `Content-type`. `Content-type` is actually sent by the response, so it's not a meaningful request header.

So basically, what we want is the following behavior:
* For all endpoints, by default, the `Accept` header is set to `application/json`
* If an endpoint defines other content types (check the specs, in TRS most of them allow `text/plain` and the `GET .../files` endpoint instead allows `application/zip`), then a corresponding `accept` parameter should be defined for the access method
* When an `accept` parameter is available for a given method, the value provided by the user should be checked against the values defined in the specs, and an error should be returned if the user asks for a content type that the specs do not provide (e.g., user passes `application/octet-stream` to `accept`, but the endpoint definition only mentions `application/json` and `text/plain`, this should return an `InvalidContentType` response).

Reply:
Do you mean that the function signature should be now:
```
def get_descriptor(
        self,
        type: str,
        id: str,
        version_id: str,
        token: Optional[str] = None,
        accept: str = "application/json"
    )
```
Right?

But I feel we are making things complex for the user who is using the client. 

I mean we can surely check on the request that is sent to the server using `Content-type` (maybe this is also not necessary), but we can accept whatever is sent to us by the server. If we keep the Header:  `'Accept': '*/*'`, will it cause any problem?

Reply:
The thing is that the client can request different responses by selecting different Accept headers. This is not something connexion takes care of

Reply:
In other words: it's really the point that we do _not_ accept what the server sends us if we want, say, a ZIP file rather than JSON. Check up on content negotiation

Reply:
Signature looks good except that version ID should be optional